src/include/network/routines.rb

```diff
@@ -742,24 +756,62 @@ def SetIfaceDown(dev_name)
 
     # Tries to set all interfaces up
     #
-    # @return [boolean] false if some of interfaces cannot be set up
+    # @return [boolean] false if interfaces are empty or some
+    # interface cannot be bring up
     def SetAllIfacesUp
+      all_up = true
       interfaces = GetAllInterfaces()
 
       return false if interfaces.empty?
 
-      interfaces.all? { |i| SetIfaceUp(i) }
+      interfaces.each { |i| all_up = false if !SetIfaceUp(i) }
+
+      all_up
```

Comment:
OK, this is a difference in fixing a maintenance branch and cleaning up for master. I now see that of the 8 methods (All or Single, Interface or Link, Up or Down), only 3 are used, but the others could be kept for symmetry. If we make up our minds about the return values (see my other comment) they should become more readable along the way.